app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_comments(id)
+    Comment.where(post_id: id).last(5)
+  end
+
+  def update_counter(id)
+    user = User.find_by(id: id)
+    user.posts_counter = Post.where(user_id: id).count
+    user.save
+  end
```

Comment:
### { Optional }

- [ ] A more efficient way to solve this is by `author.increment!(:posts_counter)` remember we have already pointed out that the user should be an author.
### Hint:

- [ ] You can even use `after_save` method and then call your update counter method. And also remember there is already an association. So there is no need to query the User again. The same idea can be used in the `like` model